OdpDescriptor.cs

```diff
@@ -144,12 +144,13 @@ public override void SetParametersOnCommand(IDbCommand cmd, IEnumerable paramete
 
                 string paramName = DbCaches.ParamNames.Get(x);
                 
-                OracleParameter oracleParam;
-                if (addMe != DBNull.Value && addMe.GetType().GetProperty("GeometryType") != null)
-                {
-                    oracleParam = new OracleParameter(paramName, OracleDbType.Object, addMe.ToString().Length, addMe.ToString());
-                    oracleParam.UdtTypeName = "SDE.ST_GEOMETRY";
-                }
+                OracleParameter oracleParam;
+                if (addMe != DBNull.Value && addMe is string && ((string)addMe).Length > 3999)
```

Comment:
The results of the query shows that the length can be pretty small, all the way down to 128 characters. That seems to low to set this to, but I'm also not sure we should mess with this and just implement #7.

It's unclear what is happening to the geometry values that don't make this check. They appear to still be inserted into the table.